starttls/imap.go

```diff
@@ -0,0 +1,67 @@
+/*-
+ * Copyright 2017 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package starttls
+
+import (
+	"bufio"
+	"crypto/tls"
+	"fmt"
+	"net"
+)
+
+func dumpAuthTLSFromIMAP(dialer Dialer, address string, config *tls.Config) (*tls.ConnectionState, error) {
```

Comment:
Nit: the original naming here was probably taken from the FTP code, which sends an AUTH TLS message as part of the FTP STARTTLS protocol (explicit FTPS). Would probably make more sense if we called these something else now, like maybe `dumpTLSConnStateFromFTP` and `dumpTLSConnStateFromIMAP`.